### PR TITLE
Use Django file caching for .json urls. Fixes #341

### LINKED ISSUE #341: Cache json output from /api
./manage.py generate_api_json
./manage.py collectstatic

if DEBUG=True, get it from Django, otherwise from /static/

### PR DESCRIPTION
**Pull request by andreeadima**
@melish please review 😄

### CODE REVIEW

File: dv/urls/api.py [hunk 1]
```diff
@@ -1,27 +1,41 @@
...
+from django.views.decorators.cache import cache_page
+
 from dv.views import api as views
 from dv.views import frontend as front_views
 
 
 urlpatterns = [
     # TODO Add this back if you want to test api/grants.json without changing the UI.
     # TODO Remove them after the change
-    # url(r'^allocation.csv', views.beneficiaries_fm_gross_allocation,
```
Comment (melish) [inline dv/urls/api.py]: We don't need the csv's, they were used during prototyping phase. Please remove

File: dv/urls/api.py [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+        name='grants'),
+    url(r'^projects.json',
+        cache_page(settings.API_CACHE_SECONDS)(views.projects),
+        name='projects'),
+    url(r'^partners.json',
+        cache_page(settings.API_CACHE_SECONDS)(views.partners),
+        name='partners'),
     url(r'^grants/(?P<beneficiary>[A-Z]{2}).json',
-        views.beneficiary_detail,
+        cache_page(settings.API_CACHE_SECONDS)(views.beneficiary_detail),
```
Comment (melish) [inline dv/urls/api.py]: We need a simple way to disable caching during development. How does it work?

File: dv/settings.py [hunk 3]
```diff
@@ -181,4 +185,15 @@
     },
 }
 
+# Caching system
+
+CACHES = {
+    'default': {
+        'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+        'LOCATION': '/var/tmp/django_cache',
```
Comment (melish) [inline dv/settings.py]: manage.py import and manage.py import_news should clear this dir

File: dv/urls/api.py (hunk 2 above)
Comment (andreeadima) [reply dv/urls/api.py]: Added to localsettings.py.example. It can be disabled by using a dummy caching backend.

File: dv/settings.py (hunk 3 above)
Comment (andreeadima) [reply dv/settings.py]: Caching cleared.

**Comment by xlotlu** [comment]
@andreeadima, @melish caching should happen at web-server level (and browser-level), not at django-level, since each import renders a stable, unique, non-dynamic output.

django should only take care of:
- a. cache busting: make sure there's a unique hash in the url that corresponds to the current import. the unix epoch transformed to base62 is nice enough (or random, there's a function in utils): `/api/1dxZ8y/grants.json`.
the hash can be generated on import and simply written into a file
- b. make sure response headers are cache-friendly. i.e. no session cookies are set for `/api/*`, and there aren't any `Vary` headers. 

the web server can then be configured to forcefully cache anything `/api/` for all eternity.

**Pull request merged**